### ISSUE TITLE
Issues with val_loss in Checkpoint Path During TensorFlow Model Training

### ISSUE THREAD

**Issue by varshad18** [keras-team/keras#19339]
I'm encountering a challenge while training a model in TensorFlow and would appreciate insights from the community.

> Error Message: 'Failed to format this callback filepath: "..." Reason: \'val_loss\'

I'm using a callback to save checkpoints during training. The checkpoint file path includes placeholders for epoch number ({epoch:02d}) and validation loss ({val_loss:.2f}). And the above-mentioned error was encountered.

`checkpoint_dir = checkpoint_dir + "/checkpoint-{epoch:02d}-{val_loss:.2f}.tf"`
### Additional info:

**COMPILE FUNCTION CODE:**

```
#Model Compile and setup with checkpoint call back


def compileSetupModel(model, checkpoint_dir = "/content/drive/MyDrive/FashionBody/Regression/TrainingRun/Run4"):
  print("Contents of trainingRun checkpoints directory ")
  !ls $checkpoint_dir
  model.compile(
    optimizer=tf.keras.optimizers.SGD(learning_rate=0.0005, momentum=0.9),
    loss = tf.keras.losses.MeanAbsoluteError())


checkpoint_dir = checkpoint_dir + "/checkpoint-{epoch:02d}-{val_loss:.2f}.tf"
mc = tf.keras.callbacks.ModelCheckpoint(filepath=checkpoint_dir, save_best_only=True, verbose=1, save_freq='epoch')   
 #callback for tensorboard
 tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1)
 return mc, tensorboard_callback
 mc, tensorboard_callback = compileSetupModel(model)
```

**FIT FUNCTION CODE**
```
epochs = 300  
BATCH_SIZE = 4
train_size = trainImageTotalData
valid_size = validationImageTotalData


steps_per_epoch = (train_size // BATCH_SIZE)-1
validation_steps = (valid_size // BATCH_SIZE)-1
hist = model.fit(
    x=[trainNumericData, trainImagesSBData, trainImagesCBData, trainImagesWBData, trainImagesHBData,trainImagesLLData,trainImagesLBData, trainImagesUpLeftABData,trainImagesUpRightABData, trainImagesALeftLData, trainImagesARightLData],   #x=trainImages,    # images
    y=trainAllRegressionData,  #y=train["severity"],  # severity target regression value
    epochs=epochs,steps_per_epoch=steps_per_epoch,
    validation_data=([validationNumericData, validationImagesSBData, validationImagesCBData, validationImagesWBData, validationImagesHBData, validationImagesLLData,validationImagesLBData, validationImagesUpLeftABData, validationImagesUpRightABData, validationImagesALeftLData, validationImagesARightLData], validationAllRegressionData),
    validation_steps=validation_steps,
    callbacks=[mc, tensorboard_callback]).history   #callback for both checkpoints and tensorboard.



```

**Comment by sachinprasadhs** [comment]
Hi, Looks like you are still using Keras 2, could you please migrate your code to Keras 3 and let us know the outcome.

https://keras.io/guides/migrating_to_keras_3/
